// @material-tailwind/react/components/progress.cjs
"use strict";
"use client";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/progress.tsx
var progress_exports = {};
__export(progress_exports, {
  Progress: () => Progress,
  ProgressBar: () => ProgressBar,
  ProgressContext: () => ProgressContext,
  ProgressRoot: () => ProgressRoot,
  default: () => progress_default
});
module.exports = __toCommonJS(progress_exports);
var React2 = __toESM(require("react"), 1);

// src/context/index.tsx
var React = __toESM(require("react"), 1);

// src/theme/accordion.ts
var accordionTheme = {
  defaultProps: {
    type: "single"
  }
};
var accordionItemTheme = {
  defaultProps: {
    disabled: false
  },
  baseStyle: "group block w-full items-center justify-between cursor-pointer border-b border-surface aria-disabled:opacity-50 aria-disabled:pointer-events-none aria-disabled:select-none"
};
var accordionTriggerTheme = {
  baseStyle: "flex items-center justify-between w-full py-3 text-left font-medium dark:text-white text-black"
};
var accordionContentTheme = {
  baseStyle: "block pb-3 text-foreground h-0 data-[open=true]:h-auto"
};

// src/theme/alert.ts
var alertTheme = {
  defaultProps: {
    size: "md",
    color: "primary",
    variant: "solid",
    rounded: false
  },
  baseStyle: "relative flex items-start w-full border rounded-md p-2 data-[pill=true]:rounded-full",
  variant: {
    ghost: {
      primary: "bg-primary/10 border-transparent text-primary",
      secondary: "bg-secondary/10 border-transparent text-secondary-foreground",
      info: "bg-info/10 border-transparent text-info",
      success: "bg-success/10 border-transparent text-success",
      warning: "bg-warning/10 border-transparent text-warning",
      error: "bg-error/10 border-transparent text-error"
    },
    solid: {
      primary: "bg-primary border-primary text-primary-foreground",
      secondary: "bg-secondary border-secondary text-secondary-foreground",
      info: "bg-info border-info text-info-foreground",
      success: "bg-success border-success text-success-foreground",
      warning: "bg-warning border-warning text-warning-foreground",
      error: "bg-error border-error text-error-foreground"
    },
    outline: {
      primary: "bg-transparent border-primary text-primary",
      secondary: "bg-transparent border-secondary text-secondary-foreground",
      info: "bg-transparent border-info text-info",
      success: "bg-transparent border-success text-success",
      warning: "bg-transparent border-warning text-warning",
      error: "bg-transparent border-error text-error"
    },
    gradient: {
      primary: "bg-gradient-to-tr from-primary-dark to-primary-light border-primary text-primary-foreground",
      secondary: "bg-gradient-to-tr from-secondary-dark to-secondary-light border-secondary text-secondary-foreground",
      info: "bg-gradient-to-tr from-info-dark to-info-light border-info text-info-foreground",
      success: "bg-gradient-to-tr from-success-dark to-success-light border-success text-success-foreground",
      warning: "bg-gradient-to-tr from-warning-dark to-warning-light border-warning text-warning-foreground",
      error: "bg-gradient-to-tr from-error-dark to-error-light border-error text-error-foreground"
    }
  }
};
var alertContentTheme = {
  baseStyle: "w-full text-base font-sans  leading-none m-1.5"
};
var alertIconTheme = {
  baseStyle: "grid place-items-center shrink-0 p-1"
};
var alertDismissTriggerTheme = {
  baseStyle: "outline-none"
};

// src/theme/avatar.ts
var avatarTheme = {
  defaultProps: {
    size: "md",
    variant: "circular"
  },
  baseStyle: "inline-block object-cover object-center data-[shape=square]:rounded-none data-[shape=circular]:rounded-full data-[shape=rounded]:rounded-[current]",
  size: {
    xs: "w-6 h-6 rounded-sm",
    sm: "w-8 h-8 rounded",
    md: "w-11 h-11 rounded-md",
    lg: "w-14 h-14 rounded-md",
    xl: "w-[72px] h-[72px] rounded-lg",
    xxl: "w-[86px] h-[86px] rounded-xl"
  }
};

// src/theme/breadcrumb.ts
var breadcrumbTheme = {
  baseStyle: "flex flex-wrap items-center gap-1 p-1"
};
var breadcrumbLinkTheme = {
  baseStyle: "inline-flex items-center gap-1.5 text-sm text-black dark:text-white transition-colors duration-300 ease-in hover:text-primary dark:hover:text-primary"
};
var breadcrumbSeparatorTheme = {
  baseStyle: "inline-block mx-1 text-sm select-none pointer-events-none opacity-50 text-black dark:text-white"
};

// src/theme/button.ts
var buttonTheme = {
  defaultProps: {
    size: "md",
    color: "primary",
    variant: "solid",
    ripple: true,
    isPill: false,
    isFullWidth: false
  },
  baseStyle: "inline-flex items-center justify-center border align-middle select-none font-sans font-medium text-center transition-all duration-300 ease-in disabled:opacity-50 disabled:shadow-none disabled:cursor-not-allowed data-[shape=pill]:rounded-full data-[width=full]:w-full",
  size: {
    sm: "text-sm leading-none py-1.5 px-2.5 rounded-md",
    md: "text-sm leading-none py-2.5 px-3.5 rounded-md",
    lg: "text-base leading-none py-3.5 px-5 rounded-lg"
  },
  variant: {
    ghost: {
      primary: "bg-transparent border-transparent text-primary hover:bg-primary/5 hover:border-primary/5",
      secondary: "bg-transparent border-transparent text-secondary-foreground hover:bg-secondary/10 hover:border-secondary/10",
      info: "bg-transparent border-transparent text-info hover:bg-info/10 hover:border-info/10",
      success: "bg-transparent border-transparent text-success hover:bg-success/10 hover:border-success/10",
      warning: "bg-transparent border-transparent text-warning hover:bg-warning/10 hover:border-warning/10",
      error: "bg-transparent border-transparent text-error hover:bg-error/10 hover:border-error/10"
    },
    solid: {
      primary: "bg-primary border-primary text-primary-foreground hover:bg-primary-light hover:border-primary-light",
      secondary: "bg-secondary border-secondary text-secondary-foreground hover:bg-secondary-light hover:border-secondary-light",
      info: "bg-info border-info text-info-foreground hover:bg-info-light hover:border-info-light",
      success: "bg-success border-success text-success-foreground hover:bg-success-light hover:border-success-light",
      warning: "bg-warning border-warning text-warning-foreground hover:bg-warning-light hover:border-warning-light",
      error: "bg-error border-error text-error-foreground hover:bg-error-light hover:border-error-light"
    },
    outline: {
      primary: "bg-transparent border-primary text-primary hover:bg-primary hover:text-primary-foreground",
      secondary: "bg-transparent border-secondary text-secondary-foreground hover:bg-secondary",
      info: "bg-transparent border-info text-info hover:bg-info hover:text-info-foreground",
      success: "bg-transparent border-success text-success hover:bg-success hover:text-success-foreground",
      warning: "bg-transparent border-warning text-warning hover:bg-warning hover:text-warning-foreground",
      error: "bg-transparent border-error text-error hover:bg-error hover:text-error-foreground"
    },
    gradient: {
      primary: "bg-gradient-to-tr from-primary-dark to-primary-light border-primary text-primary-foreground hover:brightness-105",
      secondary: "bg-gradient-to-tr from-secondary-dark to-secondary-light border-secondary text-secondary-foreground hover:brightness-105",
      info: "bg-gradient-to-tr from-info-dark to-info-light border-info text-info-foreground hover:brightness-105",
      success: "bg-gradient-to-tr from-success-dark to-success-light border-success text-success-foreground hover:brightness-105",
      warning: "bg-gradient-to-tr from-warning-dark to-warning-light border-warning text-warning-foreground hover:brightness-105",
      error: "bg-gradient-to-tr from-error-dark to-error-light border-error text-error-foreground hover:brightness-105"
    }
  }
};

// src/theme/button-group.ts
var buttonGroupTheme = {
  defaultProps: {
    size: "md",
    color: "primary",
    variant: "solid",
    ripple: true,
    isFullWidth: false,
    isPill: false,
    orientation: "horizontal"
  },
  baseStyle: "inline-flex data-[width=full]:w-full data-[orientation=horizontal]:flex-row data-[orientation=horizontal]:[&:not(:first-child):not(:last-child):not([data-variant=ghost])]:[&_*]:rounded-none data-[orientation=horizontal]:[&:first-child:not([data-variant=ghost])]:[&_*]:rounded-r-none data-[orientation=horizontal]:[&:last-child:not([data-variant=ghost])]:[&_*]:rounded-l-none data-[orientation=horizontal]:[&:not(:first-child):not(:last-child):not([data-variant=ghost])]:[&_*]:border-l-0 data-[orientation=horizontal]:[&:last-child:not([data-variant=ghost])]:[&_*]:border-l-0 data-[orientation=vertical]:flex-col data-[orientation=vertical]:[&:not(:first-child):not(:last-child):not([data-variant=ghost])]:[&_*]:rounded-none data-[orientation=vertical]:[&:first-child:not([data-variant=ghost])]:[&_*]:rounded-b-none data-[orientation=vertical]:[&:last-child:not([data-variant=ghost])]:[&_*]:rounded-t-none data-[orientation=vertical]:[&:not(:first-child):not(:last-child):not([data-variant=ghost])]:[&_*]:border-t-0 data-[orientation=vertical]:[&:last-child:not([data-variant=ghost])]:[&_*]:border-t-0"
};

// src/theme/checkbox.ts
var checkboxTheme = {
  defaultProps: {
    color: "primary"
  },
  baseStyle: "inline-block relative h-5 w-5 rounded bg-surface transition-all duration-200 ease-in aria-disabled:opacity-50 aria-disabled:pointer-events-none",
  color: {
    primary: "data-[checked=true]:bg-primary text-primary-foreground",
    secondary: "data-[checked=true]:bg-secondary text-secondary-foreground",
    info: "data-[checked=true]:bg-info text-info-foreground",
    success: "data-[checked=true]:bg-success text-success-foreground",
    warning: "data-[checked=true]:bg-warning text-warning-foreground",
    error: "data-[checked=true]:bg-error text-error-foreground"
  }
};
var checkboxIndicatorTheme = {
  baseStyle: "pointer-events-none absolute left-2/4 top-2/4 text-current -translate-x-2/4 -translate-y-2/4 scale-75 opacity-0 transition-all duration-200 ease-in data-[checked=true]:scale-100 data-[checked=true]:opacity-100"
};

// src/theme/card.ts
var cardTheme = {
  defaultProps: {
    variant: "solid",
    color: "default"
  },
  baseStyle: "w-full rounded-lg border shadow-lg overflow-hidden",
  variant: {
    ghost: {
      default: "bg-background/10 border-transparent shadow-transparent",
      primary: "bg-primary/10 border-transparent shadow-transparent",
      secondary: "bg-secondary/10 border-transparent shadow-transparent",
      info: "bg-info/10 border-transparent shadow-transparent",
      success: "bg-success/10 border-transparent shadow-transparent",
      warning: "bg-warning/10 border-transparent shadow-transparent",
      error: "bg-error/10 border-transparent shadow-transparent"
    },
    solid: {
      default: "bg-background border-surface shadow-black/5",
      primary: "bg-primary border-primary-dark shadow-primary-dark/25",
      secondary: "bg-secondary border-secondary-dark shadow-secondary-dark/25",
      info: "bg-info border-info-dark shadow-info-dark/25",
      success: "bg-success border-success-dark shadow-success-dark/25",
      warning: "bg-warning border-warning-dark shadow-warning-dark/25",
      error: "bg-error border-error-dark shadow-error-dark/25"
    },
    outline: {
      default: "bg-transparent border-surface shadow-transparent",
      primary: "bg-transparent border-primary shadow-transparent",
      secondary: "bg-transparent border-secondary shadow-transparent",
      info: "bg-transparent border-info shadow-transparent",
      success: "bg-transparent border-success shadow-transparent",
      warning: "bg-transparent border-warning shadow-transparent",
      error: "bg-transparent border-error shadow-transparent"
    },
    gradient: {
      default: "bg-background border-surface shadow-black/5",
      primary: "bg-gradient-to-t from-primary-dark to-primary-light shadow-primary/25 border-primary",
      secondary: "bg-gradient-to-t from-secondary-dark to-secondary-light shadow-secondary/25 border-secondary",
      info: "bg-gradient-to-t from-info-dark to-info-light shadow-info/25 border-info",
      success: "bg-gradient-to-t from-success-dark to-success-light shadow-success/25 border-success",
      warning: "bg-gradient-to-t from-warning-dark to-warning-light shadow-warning/25 border-warning",
      error: "bg-gradient-to-t from-error-dark to-error-light shadow-error/25 border-error"
    }
  }
};
var cardHeaderTheme = {
  baseStyle: "w-[calc(100%-12px)] h-max rounded-[5px] m-1.5"
};
var cardBodyTheme = {
  baseStyle: "w-full h-max rounded px-3 py-2"
};
var cardFooterTheme = {
  baseStyle: "w-full px-3 pt-1.5 pb-3 rounded"
};

// src/theme/chip.ts
var chipTheme = {
  defaultProps: {
    size: "md",
    color: "primary",
    variant: "solid"
  },
  baseStyle: "relative inline-flex items-center border select-none font-sans font-medium rounded-full",
  size: {
    sm: "text-xs leading-none p-0.5",
    md: "text-[13px] leading-none p-[3px]",
    lg: "text-[15px] leading-none p-1"
  },
  variant: {
    ghost: {
      primary: "bg-primary/10 border-transparent text-primary",
      secondary: "bg-secondary/10 border-transparent text-secondary-foreground",
      info: "bg-info/10 border-transparent text-info",
      success: "bg-success/10 border-transparent text-success",
      warning: "bg-warning/10 border-transparent text-warning",
      error: "bg-error/10 border-transparent text-error"
    },
    solid: {
      primary: "bg-primary border-primary text-primary-foreground",
      secondary: "bg-secondary border-secondary text-secondary-foreground",
      info: "bg-info border-info text-info-foreground",
      success: "bg-success border-success text-success-foreground",
      warning: "bg-warning border-warning text-warning-foreground",
      error: "bg-error border-error text-error-foreground"
    },
    outline: {
      primary: "bg-transparent border-primary text-primary",
      secondary: "bg-transparent border-secondary text-secondary-foreground",
      info: "bg-transparent border-info text-info",
      success: "bg-transparent border-success text-success",
      warning: "bg-transparent border-warning text-warning",
      error: "bg-transparent border-error text-error"
    },
    gradient: {
      primary: "bg-gradient-to-tr from-primary-dark to-primary-light border-primary text-primary-foreground",
      secondary: "bg-gradient-to-tr from-secondary-dark to-secondary-light border-secondary text-secondary-foreground",
      info: "bg-gradient-to-tr from-info-dark to-info-light border-info text-info-foreground",
      success: "bg-gradient-to-tr from-success-dark to-success-light border-success text-success-foreground",
      warning: "bg-gradient-to-tr from-warning-dark to-warning-light border-warning text-warning-foreground",
      error: "bg-gradient-to-tr from-error-dark to-error-light border-error text-error-foreground"
    }
  }
};
var chipLabelTheme = {
  baseStyle: "font-inherit text-inherit leading-none",
  size: {
    sm: "my-0.5 mx-1.5",
    md: "my-[3.5px] mx-2",
    lg: "my-[4.5px] mx-2.5"
  }
};
var chipIconTheme = {
  baseStyle: "grid place-items-center shrink-0 rounded-full",
  size: {
    sm: "translate-x-px w-3.5 h-3.5",
    md: "translate-x-0.5 w-4 h-4",
    lg: "translate-x-[3px] w-5 h-5"
  }
};
var chipDismissTriggerTheme = {
  defaultProps: {
    ripple: true
  },
  baseStyle: "grid place-items-center shrink-0 rounded-full p-px",
  size: {
    sm: "-translate-x-px ms-0.5 w-4 h-4 stroke-2",
    md: "-translate-x-0.5 ms-1 w-[18px] h-[18px] stroke-2",
    lg: "-translate-x-[3px] ms-1.5 w-[22px] h-[22px] stroke-2"
  }
};

// src/theme/collapse.ts
var collapseTheme = {
  baseStyle: "block w-full h-0 overflow-hidden data-[open=true]:h-auto data-[open=true]:overflow-visible"
};

// src/theme/dialog.ts
var dialogTheme = {
  defaultProps: {
    size: "md"
  }
};
var dialogTriggerTheme = {
  baseStyle: "outline-none"
};
var dialogContentTheme = {
  defaultProps: {
    disabled: false,
    initialFocus: 0,
    returnFocus: true,
    guards: true,
    modal: false,
    visuallyHiddenDismiss: true,
    closeOnFocusOut: true,
    order: ["content"]
  },
  baseStyle: "fixed z-[9998] top-1/2 left-1/2 -translate-x-1/2 px-4 py-3 max-h-[calc(100vh-32px)] overflow-y-auto -translate-y-1/2 bg-background w-full h-full rounded-xl shadow-2xl shadow-black/5 border border-surface",
  size: {
    xs: "w-8/12 sm:w-6/12 md:w-4/12 lg:w-3/12 xl:w-2/12 h-max",
    sm: "w-9/12 sm:w-7/12 md:w-5/12 lg:w-3/12 h-max",
    md: "w-10/12 md:w-8/12 lg:w-6/12 h-max",
    lg: "w-10/12 lg:w-8/12 h-max",
    xl: "w-11/12 lg:w-10/12 h-max",
    screen: "w-full h-full rounded-none max-h-screen"
  }
};
var dialogOverlayTheme = {
  defaultProps: {
    lockScroll: true
  },
  baseStyle: "fixed inset-0 w-screen h-screen z-[9997] bg-black/50"
};
var dialogDismissTriggerTheme = {
  baseStyle: "outline-none"
};

// src/theme/drawer.ts
var drawerTriggerTheme = {
  baseStyle: "outline-none"
};
var drawerPanelTheme = {
  defaultProps: {
    disabled: false,
    initialFocus: 0,
    returnFocus: true,
    guards: true,
    modal: false,
    visuallyHiddenDismiss: true,
    closeOnFocusOut: true,
    order: ["content"],
    placement: "right"
  },
  baseStyle: "fixed z-[9998] p-4 bg-background w-80 h-80 shadow-2xl shadow-black/5 border-surface data-[placement=top]:top-0 data-[placement=top]:left-0 data-[placement=right]:top-0 data-[placement=right]:right-0 data-[placement=bottom]:bottom-0 data-[placement=bottom]:left-0 data-[placement=left]:top-0 data-[placement=left]:left-0 data-[placement=top]:w-screen data-[placement=bottom]:w-screen data-[placement=left]:h-screen data-[placement=right]:h-screen border border-surface"
};
var drawerOverlayTheme = {
  defaultProps: {
    lockScroll: true
  },
  baseStyle: "fixed inset-0 w-screen h-screen z-[9997] bg-black/50"
};
var drawerDismissTriggerTheme = {
  baseStyle: "outline-none"
};

// src/theme/icon-button.ts
var iconButtonTheme = {
  defaultProps: {
    size: "md",
    color: "primary",
    variant: "solid",
    ripple: true,
    isCircular: false
  },
  baseStyle: "inline-grid place-items-center border align-middle select-none font-sans font-medium text-center transition-all duration-300 ease-in disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none data-[shape=circular]:rounded-full",
  size: {
    sm: "text-sm leading-none min-w-[28px] min-h-[28px] rounded-md",
    md: "text-sm leading-none min-w-[36px] min-h-[36px] rounded-md",
    lg: "text-base leading-none min-w-[46px] min-h-[46px] rounded-lg"
  },
  variant: buttonTheme.variant
};

// src/theme/input.ts
var inputTheme = {
  defaultProps: {
    size: "md",
    color: "primary",
    isPill: false,
    isError: false,
    isSuccess: false
  },
  baseStyle: "relative w-full aria-disabled:cursor-not-allowed data-[shape=pill]:rounded-full",
  size: {
    sm: "text-sm leading-none rounded-md",
    md: "text-sm leading-none rounded-md",
    lg: "text-base leading-none rounded-lg"
  }
};
var inputFieldTheme = {
  baseStyle: "h-full w-full outline-none rounded-[inherit] text-inherit leading-[inherit] focus:outline-none text-black dark:text-white placeholder:text-foreground/60 bg-transparent ring-transparent border border-surface transition-all duration-300 ease-in disabled:opacity-50 disabled:pointer-events-none data-[error=true]:border-error data-[success=true]:border-success select-none",
  size: {
    sm: "py-[4.5px] px-1.5 ring data-[icon-placement=start]:ps-6 data-[icon-placement=end]:pe-6",
    md: "py-[8.5px] px-2.5 ring data-[icon-placement=start]:ps-8 data-[icon-placement=end]:pe-8",
    lg: "py-[12.25px] px-3.5 ring-4 data-[icon-placement=start]:ps-10 data-[icon-placement=end]:pe-10"
  },
  color: {
    primary: "hover:border-primary hover:ring-primary/10 focus:border-primary focus:ring-primary/10",
    secondary: "hover:border-secondary hover:ring-secondary/10 focus:border-secondary focus:ring-secondary/10",
    info: "hover:border-info hover:ring-info/10 focus:border-info focus:ring-info/10",
    success: "hover:border-success hover:ring-success/10 focus:border-success focus:ring-success/10",
    warning: "hover:border-warning hover:ring-warning/10 focus:border-warning focus:ring-warning/10",
    error: "hover:border-error hover:ring-error/10 focus:border-error focus:ring-error/10"
  }
};
var inputIconTheme = {
  defaultProps: {
    placement: "start"
  },
  baseStyle: "pointer-events-none absolute top-1/2 -translate-y-1/2 text-foreground/70 peer-hover:text-black peer-focus:text-black dark:peer-hover:text-white dark:peer-focus:text-white transition-all duration-300 ease-in overflow-hidden",
  size: {
    sm: "w-4 h-4 data-[placement=start]:left-[5px] data-[placement=end]:right-[5px]",
    md: "w-5 h-5 data-[placement=start]:left-[7px] data-[placement=end]:right-[7px]",
    lg: "w-6 h-6 data-[placement=start]:left-[9px] data-[placement=end]:right-[9px]"
  }
};

// src/theme/list.ts
var listTheme = {
  baseStyle: "flex flex-col gap-0.5 min-w-[240px]"
};
var listItemTheme = {
  defaultProps: {
    ripple: true
  },
  baseStyle: "flex items-center py-1.5 px-2.5 rounded-md align-middle select-none font-sans  transition-all duration-300 ease-in aria-disabled:opacity-50 aria-disabled:pointer-events-none bg-transparent text-foreground hover:text-black dark:hover:text-white hover:bg-surface focus:bg-surface focus:text-black dark:focus:text-white data-[selected=true]:bg-surface data-[selected=true]:text-black data-[selected=true]:text-white dark:bg-opacity-70"
};
var listItemStartTheme = {
  baseStyle: "grid place-items-center shrink-0 me-2.5"
};
var listItemEndTheme = {
  baseStyle: "grid place-items-center shrink-0 ps-2.5 ms-auto"
};

// src/theme/menu.ts
var menuTheme = {
  defaultProps: {
    placement: "bottom",
    offset: 5
  }
};
var menuTriggerTheme = {
  baseStyle: "outline-none group"
};
var menuContentTheme = {
  defaultProps: {
    disabled: false,
    initialFocus: 0,
    returnFocus: true,
    guards: true,
    modal: false,
    visuallyHiddenDismiss: true,
    closeOnFocusOut: true,
    order: ["content"]
  },
  baseStyle: "min-w-[160px] rounded-lg space-y-0.5 border border-surface bg-background p-1 text-sm text-foreground shadow-xl shadow-black/[0.025] outline-none"
};
var menuItemTheme = {
  defaultProps: {
    ripple: true,
    closeOnClick: true
  },
  baseStyle: "w-full text-start flex items-center py-1.5 px-2.5 rounded-[5px] align-middle select-none outline-none font-sans  transition-all duration-300 ease-in aria-disabled:opacity-50 aria-disabled:pointer-events-none bg-transparent text-foreground hover:text-black dark:hover:text-white hover:bg-surface focus:bg-surface focus:text-black dark:focus:text-white data-[selected=true]:bg-surface data-[selected=true]:text-black dark:data-[selected=true]:text-white dark:bg-opacity-70"
};

// src/theme/popover.ts
var popoverTheme = {
  defaultProps: {
    placement: "bottom",
    offset: 10
  }
};
var popoverTriggerTheme = {
  baseStyle: "outline-none group"
};
var popoverContentTheme = {
  defaultProps: {
    disabled: false,
    initialFocus: 0,
    returnFocus: true,
    guards: true,
    modal: false,
    visuallyHiddenDismiss: true,
    closeOnFocusOut: true,
    order: ["content"]
  },
  baseStyle: "rounded-lg border border-surface bg-background p-2.5 text-sm text-foreground shadow-xl shadow-black/[0.025] outline-none"
};
var popoverArrowTheme = {
  baseStyle: "h-3 w-3 rounded-bl-[3px] border border-[inherit] bg-inherit [clip-path:polygon(0_0,_100%_100%,_0_100%)] data-[placement^=bottom]:rotate-[135deg] data-[placement^=left]:rotate-[225deg] data-[placement^=right]:rotate-45 data-[placement^=top]:-rotate-45"
};

// src/theme/progress.ts
var progressTheme = {
  defaultProps: {
    size: "md",
    color: "primary"
  },
  baseStyle: "w-full bg-surface block rounded-full overflow-hidden",
  size: {
    sm: "h-2",
    md: "h-4",
    lg: "h-6"
  }
};
var progressBarTheme = {
  baseStyle: "h-full rounded-none",
  color: {
    primary: "bg-primary",
    secondary: "bg-secondary",
    info: "bg-info",
    success: "bg-success",
    warning: "bg-warning",
    error: "bg-error"
  }
};

// src/theme/radio.ts
var radioTheme = {
  defaultProps: {
    color: "primary",
    orientation: "vertical"
  },
  baseStyle: "flex gap-2 data-[orientation=horizontal]:items-center data-[orientation=horizontal]:flex-row data-[orientation=vertical]:flex-col data-[orientation=vertical]:items-start"
};
var radioItemTheme = {
  baseStyle: "group block relative h-5 w-5 shrink-0 rounded-full bg-surface transition-all duration-200 ease-in aria-disabled:opacity-50 aria-disabled:pointer-events-none",
  color: {
    primary: "data-[checked=true]:bg-primary text-primary-foreground",
    secondary: "data-[checked=true]:bg-secondary text-secondary-foreground",
    info: "data-[checked=true]:bg-info text-info-foreground",
    success: "data-[checked=true]:bg-success text-success-foreground",
    warning: "data-[checked=true]:bg-warning text-warning-foreground",
    error: "data-[checked=true]:bg-error text-error-foreground"
  }
};
var radioIndicatorTheme = {
  baseStyle: "pointer-events-none absolute left-2/4 top-2/4 text-current -translate-x-2/4 -translate-y-2/4 scale-75 opacity-0 transition-all duration-200 ease-in group-data-[checked=true]:scale-100 group-data-[checked=true]:opacity-100"
};

// src/theme/rating.tsx
var import_jsx_runtime = require("react/jsx-runtime");
var ratingTheme = {
  defaultProps: {
    count: 5,
    color: "primary",
    ratedIcon: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        viewBox: "0 0 24 24",
        fill: "currentColor",
        className: "h-6 w-6",
        children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
          "path",
          {
            fillRule: "evenodd",
            d: "M10.788 3.21c.448-1.077 1.976-1.077 2.424 0l2.082 5.007 5.404.433c1.164.093 1.636 1.545.749 2.305l-4.117 3.527 1.257 5.273c.271 1.136-.964 2.033-1.96 1.425L12 18.354 7.373 21.18c-.996.608-2.231-.29-1.96-1.425l1.257-5.273-4.117-3.527c-.887-.76-.415-2.212.749-2.305l5.404-.433 2.082-5.006z",
            clipRule: "evenodd"
          }
        )
      }
    ),
    unratedIcon: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        fill: "none",
        viewBox: "0 0 24 24",
        strokeWidth: 1.5,
        stroke: "currentColor",
        className: "h-6 w-6",
        children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
          "path",
          {
            strokeLinecap: "round",
            strokeLinejoin: "round",
            d: "M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
          }
        )
      }
    ),
    readonly: false
  },
  baseStyle: "inline-flex items-center [&_data-slot=icon]:w-5 [&_data-slot=icon]:h-5 [&_data-slot=icon]:text-inherit [&_data-slot=icon]:cursor-pointer",
  color: {
    primary: "text-primary",
    secondary: "text-secondary",
    info: "text-info",
    success: "text-success",
    warning: "text-warning",
    error: "text-error"
  }
};

// src/theme/select.tsx
var import_jsx_runtime2 = require("react/jsx-runtime");
var selectTheme = {
  defaultProps: {
    size: "md",
    color: "primary",
    isPill: false,
    isError: false,
    isSuccess: false,
    placement: "bottom",
    offset: 5
  }
};
var selectTriggerTheme = {
  defaultProps: {
    indicator: /* @__PURE__ */ (0, import_jsx_runtime2.jsxs)(
      "svg",
      {
        viewBox: "0 0 24 24",
        fill: "none",
        xmlns: "http://www.w3.org/2000/svg",
        color: "currentColor",
        className: "h-[1em] w-[1em] translate-x-0.5 stroke-[1.5]",
        children: [
          /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(
            "path",
            {
              d: "M17 8L12 3L7 8",
              stroke: "currentColor",
              strokeLinecap: "round",
              strokeLinejoin: "round"
            }
          ),
          /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(
            "path",
            {
              d: "M17 16L12 21L7 16",
              stroke: "currentColor",
              strokeLinecap: "round",
              strokeLinejoin: "round"
            }
          )
        ]
      }
    )
  },
  baseStyle: "flex items-center gap-4 justify-between h-max w-full outline-none focus:outline-none text-foreground bg-transparent ring-transparent border border-surface transition-all duration-300 ease-in disabled:opacity-50 disabled:pointer-events-none data-[error=true]:border-error data-[success=true]:border-success select-none text-start data-[shape=pill]:rounded-full [&_data-slot=placeholder]:text-foreground/60",
  size: {
    sm: "text-sm leading-[normal] rounded-md py-[4.5px] px-1.5 ring",
    md: "text-sm leading-[normal] rounded-md py-[8.5px] px-2.5 ring",
    lg: "text-base leading-[normal] rounded-lg py-[12.25px] px-3.5 ring-4"
  },
  color: {
    primary: "hover:border-primary hover:ring-primary/10 focus:border-primary focus:ring-primary/10 data-[open=true]:border-primary data-[open=true]:ring-primary/10",
    secondary: "hover:border-secondary hover:ring-secondary/10 focus:border-secondary focus:ring-secondary/10 data-[open=true]:border-secondary data-[open=true]:ring-secondary/10",
    info: "hover:border-info hover:ring-info/10 focus:border-info focus:ring-info/10 data-[open=true]:border-info data-[open=true]:ring-info/10",
    success: "hover:border-success hover:ring-success/10 focus:border-success focus:ring-success/10 data-[open=true]:border-success data-[open=true]:ring-success/10",
    warning: "hover:border-warning hover:ring-warning/10 focus:border-warning focus:ring-warning/10 data-[open=true]:border-warning data-[open=true]:ring-warning/10",
    error: "hover:border-error hover:ring-error/10 focus:border-error focus:ring-error/10 data-[open=true]:border-error data-[open=true]:ring-error/10"
  }
};
var selectListTheme = {
  defaultProps: {
    disabled: false,
    initialFocus: 0,
    returnFocus: true,
    guards: true,
    modal: true,
    visuallyHiddenDismiss: true,
    closeOnFocusOut: true,
    order: ["content"]
  },
  baseStyle: "flex flex-col gap-0.5 bg-background p-1 rounded-lg shadow-xl shadow-black/[0.025] border border-surface outline-none"
};
var selectOptionTheme = {
  defaultProps: {
    ripple: true,
    indicator: /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        fill: "none",
        viewBox: "0 0 24 24",
        strokeWidth: 2,
        stroke: "currentColor",
        className: "h-4 w-4",
        children: /* @__PURE__ */ (0, import_jsx_runtime2.jsx)(
          "path",
          {
            strokeLinecap: "round",
            strokeLinejoin: "round",
            d: "M4.5 12.75l6 6 9-13.5"
          }
        )
      }
    )
  },
  baseStyle: "outline-none flex items-center justify-between gap-4 py-1.5 px-2.5 rounded-[5px] align-middle select-none text-sm font-sans transition-all duration-300 ease-in disabled:opacity-50 disabled:cursor-not-allowed bg-transparent text-foreground hover:text-black dark:hover:text-white hover:bg-surface focus:bg-surface focus:text-black dark:focus:text-white data-[selected=true]:bg-surface data-[selected=true]:text-black dark:data-[selected=true]:text-white dark:bg-opacity-70"
};

// src/theme/spinner.ts
var spinnerTheme = {
  defaultProps: {
    size: "md",
    color: "primary"
  },
  baseStyle: "text-surface animate-spin",
  size: {
    xs: "h-3 w-3",
    sm: "h-4 w-4",
    md: "h-6 w-6",
    lg: "h-8 w-8",
    xl: "h-10 w-10",
    xxl: "h-12 w-12"
  },
  color: {
    primary: "text-primary",
    secondary: "text-secondary",
    info: "text-info",
    success: "text-success",
    warning: "text-warning",
    error: "text-error"
  }
};

// src/theme/switch.ts
var switchTheme = {
  defaultProps: {
    color: "primary"
  },
  baseStyle: "appearance-none relative inline-block rounded-full w-10 h-6 before:inline-block before:absolute before:top-0 before:left-0 before:w-full before:h-full before:rounded-full before:bg-surface before:transition-colors before:duration-200 before:ease-in after:absolute after:top-2/4 after:left-0.5 after:-translate-y-2/4 after:w-5 after:h-5 after:bg-background after:rounded-full checked:after:translate-x-[calc(100%-4px)] after:transition-transform after:duration-200 after:ease-in disabled:opacity-50 disabled:cursor-not-allowed dark:after:bg-white",
  color: {
    primary: "checked:before:bg-primary",
    secondary: "checked:before:bg-secondary",
    info: "checked:before:bg-info",
    success: "checked:before:bg-success",
    warning: "checked:before:bg-warning",
    error: "checked:before:bg-error"
  }
};

// src/theme/tabs.ts
var tabsTheme = {
  defaultProps: {
    orientation: "horizontal"
  },
  baseStyle: "flex data-[orientation=horizontal]:flex-col data-[orientation=vertical]:flex-row gap-2"
};
var tabsListTheme = {
  baseStyle: "flex shrink-0 w-max data-[orientation=horizontal]:flex-row data-[orientation=vertical]:flex-col rounded-md p-1 bg-surface-light dark:bg-surface"
};
var tabsTriggerTheme = {
  baseStyle: "inline-flex relative z-[2] py-1.5 px-4 items-center justify-center align-middle text-black dark:text-white select-none font-sans font-medium text-center aria-disabled:opacity-50 aria-disabled:pointer-events-none"
};
var tabsPanelTheme = {
  baseStyle: "p-1 w-full block"
};
var tabsTriggerIndicatorTheme = {
  baseStyle: "bg-background rounded shadow-sm shadow-black/10 transition-all duration-300 ease-in"
};

// src/theme/textarea.ts
var textareaTheme = {
  defaultProps: {
    size: "md",
    color: "primary",
    resize: false,
    isError: false,
    isSuccess: false
  },
  baseStyle: "peer block w-full resize-none outline-none focus:outline-none text-black dark:text-white placeholder:text-foreground/60 bg-transparent ring-transparent border border-surface transition-all duration-300 ease-in disabled:opacity-50 disabled:pointer-events-none data-[error=true]:border-error data-[success=true]:border-success data-[resize=true]:resize-y",
  color: {
    primary: "hover:border-primary hover:ring-primary/10 focus:border-primary focus:ring-primary/10",
    secondary: "hover:border-secondary hover:ring-secondary/10 focus:border-secondary focus:ring-secondary/10",
    info: "hover:border-info hover:ring-info/10 focus:border-info focus:ring-info/10",
    success: "hover:border-success hover:ring-success/10 focus:border-success focus:ring-success/10",
    warning: "hover:border-warning hover:ring-warning/10 focus:border-warning focus:ring-warning/10",
    error: "hover:border-error hover:ring-error/10 focus:border-error focus:ring-error/10"
  },
  size: {
    sm: "text-sm leading-none p-1.5 rounded-md ring",
    md: "text-sm leading-none p-2.5 rounded-md ring",
    lg: "text-base leading-none p-3.5 rounded-lg ring-4"
  }
};

// src/theme/tooltip.ts
var tooltipTheme = {
  defaultProps: {
    placement: "top",
    offset: 8,
    interactive: false
  }
};
var tooltipTriggerTheme = {
  baseStyle: "outline-none group"
};
var tooltipContentTheme = {
  baseStyle: "rounded-md bg-black dark:bg-white border-black dark:border-white px-1.5 pt-1 pb-[3px] text-xs text-white dark:text-black shadow-lg shadow-black/5 outline-none"
};
var tooltipArrowTheme = {
  baseStyle: "h-2 w-2 rounded-bl-[3px] border border-[inherit] bg-inherit [clip-path:polygon(0_0,_100%_100%,_0_100%)] data-[placement^=bottom]:rotate-[135deg] data-[placement^=left]:rotate-[225deg] data-[placement^=right]:rotate-45 data-[placement^=top]:-rotate-45"
};

// src/theme/typography.ts
var typographyTheme = {
  defaultProps: {
    type: "p",
    color: "inherit"
  },
  baseStyle: "font-sans antialiased",
  color: {
    inherit: "text-inherit",
    default: "text-black dark:text-white",
    primary: "text-primary",
    secondary: "text-secondary",
    info: "text-info",
    success: "text-success",
    warning: "text-warning",
    error: "text-error"
  },
  type: {
    h1: "font-bold text-4xl md:text-5xl lg:text-6xl",
    h2: "font-bold text-3xl md:text-4xl lg:text-5xl",
    h3: "font-bold text-2xl md:text-3xl lg:text-4xl",
    h4: "font-bold text-xl md:text-2xl lg:text-3xl",
    h5: "font-bold text-lg md:text-xl lg:text-2xl",
    h6: "font-bold text-base md:text-lg lg:text-xl",
    lead: " text-base md:text-lg",
    p: " text-base",
    small: " text-sm"
  }
};

// src/theme/timeline.ts
var timelineTheme = {
  defaultProps: {
    color: "primary",
    mode: "timeline",
    orientation: "horizontal"
  },
  baseStyle: "flex w-full data-[orientation=horizontal]:flex-row data-[orientation=vertical]:flex-col data-[orientation=vertical]:items-start data-[orientation=horizontal]:items-center"
};
var timelineItemTheme = {
  baseStyle: "group data-[orientation=horizontal]:block data-[orientation=vertical]:flex data-[orientation=vertical]:gap-x-6 aria-disabled:select-none aria-disabled:pointer-events-none"
};
var timelineHeaderTheme = {
  baseStyle: "relative"
};
var timelineIconTheme = {
  baseStyle: "relative z-10 grid h-10 w-10 place-items-center rounded-full bg-surface text-surface-foreground",
  color: {
    primary: "group-data-[active=true]:bg-primary group-data-[active=true]:text-primary-foreground group-data-[completed=true]:bg-primary group-data-[completed=true]:text-primary-foreground",
    secondary: "group-data-[active=true]:bg-secondary group-data-[active=true]:text-secondary-foreground group-data-[completed=true]:bg-secondary group-data-[completed=true]:text-secondary-foreground",
    info: "group-data-[active=true]:bg-info group-data-[active=true]:text-info-foreground group-data-[completed=true]:bg-info group-data-[completed=true]:text-info-foreground",
    success: "group-data-[active=true]:bg-success group-data-[active=true]:text-success-foreground group-data-[completed=true]:bg-success group-data-[completed=true]:text-success-foreground",
    warning: "group-data-[active=true]:bg-warning group-data-[active=true]:text-warning-foreground group-data-[completed=true]:bg-warning group-data-[completed=true]:text-warning-foreground",
    error: "group-data-[active=true]:bg-error group-data-[active=true]:text-error-foreground group-data-[completed=true]:bg-error group-data-[completed=true]:text-error-foreground"
  }
};
var timelineSeparatorTheme = {
  baseStyle: "bg-surface absolute data-[orientation=vertical]:left-1/2 data-[orientation=vertical]:top-0 data-[orientation=vertical]:h-full data-[orientation=vertical]:w-0.5 data-[orientation=vertical]:-translate-x-1/2 data-[orientation=horizontal]:top-1/2 data-[orientation=horizontal]:left-0 data-[orientation=horizontal]:h-0.5 data-[orientation=horizontal]:w-full data-[orientation=horizontal]:-translate-y-1/2",
  color: {
    primary: "group-data-[completed=true]:bg-primary",
    secondary: "group-data-[completed=true]:bg-secondary",
    info: "group-data-[completed=true]:bg-info",
    success: "group-data-[completed=true]:bg-success",
    warning: "group-data-[completed=true]:bg-warning",
    error: "group-data-[completed=true]:bg-error"
  }
};
var timelineBodyTheme = {
  baseStyle: "data-[orientation=vertical]:pb-8 data-[orientation=horizontal]:py-4 text-foreground"
};

// src/theme/theme.ts
var theme = {
  accordion: accordionTheme,
  accordionItem: accordionItemTheme,
  accordionTrigger: accordionTriggerTheme,
  accordionContent: accordionContentTheme,
  alert: alertTheme,
  alertContent: alertContentTheme,
  alertIcon: alertIconTheme,
  alertDismissTrigger: alertDismissTriggerTheme,
  avatar: avatarTheme,
  breadcrumb: breadcrumbTheme,
  breadcrumbLink: breadcrumbLinkTheme,
  breadcrumbSeparator: breadcrumbSeparatorTheme,
  button: buttonTheme,
  buttonGroup: buttonGroupTheme,
  checkbox: checkboxTheme,
  checkboxIndicator: checkboxIndicatorTheme,
  card: cardTheme,
  cardHeader: cardHeaderTheme,
  cardBody: cardBodyTheme,
  cardFooter: cardFooterTheme,
  chip: chipTheme,
  chipLabel: chipLabelTheme,
  chipIcon: chipIconTheme,
  chipDismissTrigger: chipDismissTriggerTheme,
  collapse: collapseTheme,
  dialog: dialogTheme,
  dialogTrigger: dialogTriggerTheme,
  dialogOverlay: dialogOverlayTheme,
  dialogContent: dialogContentTheme,
  dialogDismissTrigger: dialogDismissTriggerTheme,
  drawerTrigger: drawerTriggerTheme,
  drawerOverlay: drawerOverlayTheme,
  drawerPanel: drawerPanelTheme,
  drawerDismissTrigger: drawerDismissTriggerTheme,
  iconButton: iconButtonTheme,
  input: inputTheme,
  inputField: inputFieldTheme,
  inputIcon: inputIconTheme,
  list: listTheme,
  listItem: listItemTheme,
  listItemStart: listItemStartTheme,
  listItemEnd: listItemEndTheme,
  menu: menuTheme,
  menuContent: menuContentTheme,
  menuTrigger: menuTriggerTheme,
  menuItem: menuItemTheme,
  popover: popoverTheme,
  popoverTrigger: popoverTriggerTheme,
  popoverContent: popoverContentTheme,
  popoverArrow: popoverArrowTheme,
  progress: progressTheme,
  progressBar: progressBarTheme,
  radio: radioTheme,
  radioItem: radioItemTheme,
  radioIndicator: radioIndicatorTheme,
  rating: ratingTheme,
  select: selectTheme,
  selectTrigger: selectTriggerTheme,
  selectList: selectListTheme,
  selectOption: selectOptionTheme,
  spinner: spinnerTheme,
  switch: switchTheme,
  tabs: tabsTheme,
  tabsList: tabsListTheme,
  tabsTrigger: tabsTriggerTheme,
  tabsPanel: tabsPanelTheme,
  tabsTriggerIndicator: tabsTriggerIndicatorTheme,
  textarea: textareaTheme,
  tooltip: tooltipTheme,
  tooltipTrigger: tooltipTriggerTheme,
  tooltipContent: tooltipContentTheme,
  tooltipArrow: tooltipArrowTheme,
  typography: typographyTheme,
  timeline: timelineTheme,
  timelineItem: timelineItemTheme,
  timelineHeader: timelineHeaderTheme,
  timelineIcon: timelineIconTheme,
  timelineSeparator: timelineSeparatorTheme,
  timelineBody: timelineBodyTheme
};

// src/context/index.tsx
var import_deepmerge = __toESM(require("deepmerge"), 1);

// src/utils/merge-theme.ts
var import_tailwind_merge = require("tailwind-merge");

// src/context/index.tsx
var import_jsx_runtime3 = require("react/jsx-runtime");
var MaterialTailwindThemeContext = React.createContext(theme);
MaterialTailwindThemeContext.displayName = "MaterialTailwind.ThemeProvider";
function useTheme() {
  const context = React.useContext(MaterialTailwindThemeContext);
  if (!context) {
    return null;
  }
  return context;
}

// src/components/progress.tsx
var import_tailwind_merge2 = require("tailwind-merge");
var import_jsx_runtime4 = require("react/jsx-runtime");
var ProgressContext = React2.createContext({
  value: 0,
  color: "primary"
});
var ProgressRoot = React2.forwardRef(
  ({ as, size, color, value, className, children, ...rest }, ref) => {
    const Element = as ?? "div";
    const contextTheme = useTheme();
    const theme2 = contextTheme.progress ?? progressTheme;
    const defaultProps = theme2?.defaultProps;
    size ??= defaultProps?.size ?? "md";
    color ??= defaultProps?.color ?? "primary";
    const styles = (0, import_tailwind_merge2.twMerge)(theme2.baseStyle, theme2.size[size], className);
    const contextValue = React2.useMemo(
      () => ({ value, color }),
      [value, color]
    );
    return /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(ProgressContext.Provider, { value: contextValue, children: /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(Element, { ...rest, ref, className: styles, children }) });
  }
);
ProgressRoot.displayName = "MaterialTailwind.Progress";
var ProgressBar = React2.forwardRef(
  ({ as, className, children, ...rest }, ref) => {
    const Element = as ?? "div";
    const contextTheme = useTheme();
    const { color, value } = React2.useContext(ProgressContext);
    const theme2 = contextTheme.progressBar ?? progressBarTheme;
    const styles = (0, import_tailwind_merge2.twMerge)(
      theme2.baseStyle,
      theme2.color[color],
      className
    );
    return /* @__PURE__ */ (0, import_jsx_runtime4.jsx)(
      Element,
      {
        ...rest,
        ref,
        className: styles,
        style: {
          width: `${value}%`,
          ...rest?.style
        },
        children
      }
    );
  }
);
ProgressBar.displayName = "MaterialTailwind.ProgressBar";
var Progress = Object.assign(ProgressRoot, {
  Bar: ProgressBar
});
var progress_default = Progress;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Progress,
  ProgressBar,
  ProgressContext,
  ProgressRoot
});
