// @material-tailwind/react/chunk-ZFGLD6TC.js
import {
  useTheme
} from "./chunk-KYDE6FLK.js";
import {
  cardBodyTheme,
  cardFooterTheme,
  cardHeaderTheme,
  cardTheme
} from "./chunk-ELQB6BWL.js";

// src/components/card.tsx
import * as React from "react";
import { twMerge } from "tailwind-merge";
import { jsx } from "react/jsx-runtime";
var CardRoot = React.forwardRef(
  ({ as, color, variant, className, children, ...rest }, ref) => {
    const Element = as || "div";
    const contextTheme = useTheme();
    const theme = contextTheme.card || cardTheme;
    const defaultProps = theme.defaultProps;
    color ??= defaultProps?.color ?? "default";
    variant ??= defaultProps?.variant ?? "solid";
    const styles = twMerge(
      theme.baseStyle,
      theme.variant[variant][color],
      className
    );
    return /* @__PURE__ */ jsx(Element, { ...rest, ref, className: styles, children });
  }
);
CardRoot.displayName = "MaterialTailwind.Card";
var CardHeader = React.forwardRef(
  ({ as, className, children, ...rest }, ref) => {
    const Element = as || "div";
    const contextTheme = useTheme();
    const theme = contextTheme.cardHeader || cardHeaderTheme;
    const styles = twMerge(theme.baseStyle, className);
    return /* @__PURE__ */ jsx(Element, { ...rest, ref, className: styles, children });
  }
);
CardHeader.displayName = "MaterialTailwind.CardHeader";
var CardBody = React.forwardRef(
  ({ as, className, children, ...rest }, ref) => {
    const Element = as || "div";
    const contextTheme = useTheme();
    const theme = contextTheme.cardBody || cardBodyTheme;
    const styles = twMerge(theme.baseStyle, className);
    return /* @__PURE__ */ jsx(Element, { ...rest, ref, className: styles, children });
  }
);
CardBody.displayName = "MaterialTailwind.CardBody";
var CardFooter = React.forwardRef(
  ({ as, className, children, ...rest }, ref) => {
    const Element = as || "div";
    const contextTheme = useTheme();
    const theme = contextTheme.cardFooter || cardFooterTheme;
    const styles = twMerge(theme.baseStyle, className);
    return /* @__PURE__ */ jsx(Element, { ...rest, ref, className: styles, children });
  }
);
CardFooter.displayName = "MaterialTailwind.CardFooter";
var Card = Object.assign(CardRoot, {
  Header: CardHeader,
  Body: CardBody,
  Footer: CardFooter
});
var card_default = CardFooter;

export {
  CardRoot,
  CardHeader,
  CardBody,
  CardFooter,
  Card,
  card_default
};
