// @material-tailwind/react/chunk-L35DMLEP.js
import {
  useTheme
} from "./chunk-KYDE6FLK.js";
import {
  accordionContentTheme,
  accordionItemTheme,
  accordionTheme,
  accordionTriggerTheme
} from "./chunk-T63Y5BBA.js";

// src/components/accordion.tsx
import * as React from "react";
import { twMerge } from "tailwind-merge";
import { jsx } from "react/jsx-runtime";
var AccordionContext = React.createContext({
  type: "single",
  activeItem: "",
  setActiveItem: () => {
  }
});
function AccordionRoot({
  type,
  value,
  defaultValue,
  onValueChange,
  children
}) {
  const contextTheme = useTheme();
  const theme = contextTheme.accordion || accordionTheme;
  const defaultProps = theme?.defaultProps;
  type ??= defaultProps?.type ?? "single";
  const accordionValue = value || defaultValue;
  const [uncontrolledActiveItem, setUncontrolledActiveItem] = React.useState("");
  const activeItem = value || uncontrolledActiveItem;
  const setActiveItem = onValueChange || setUncontrolledActiveItem;
  React.useEffect(() => {
    setActiveItem(accordionValue);
  }, [accordionValue]);
  const contextValue = React.useMemo(
    () => ({
      type,
      activeItem,
      setActiveItem
    }),
    [type, activeItem, setActiveItem]
  );
  return /* @__PURE__ */ jsx(AccordionContext.Provider, { value: contextValue, children });
}
AccordionRoot.displayName = "MaterialTailwind.Accordion";
var AccordionItemContext = React.createContext("");
var AccordionItem = React.forwardRef(
  ({ as, value, disabled, className, children, ...rest }, ref) => {
    const Element = as || "div";
    const contextTheme = useTheme();
    const theme = contextTheme.accordionItem || accordionItemTheme;
    const defaultProps = theme?.defaultProps;
    const { type, activeItem } = React.useContext(AccordionContext);
    disabled ??= defaultProps?.disabled ?? false;
    const isMultiple = type === "multiple";
    const isOpen = isMultiple ? activeItem?.includes(value) : activeItem === value;
    const styles = twMerge(theme.baseStyle, className);
    return /* @__PURE__ */ jsx(AccordionItemContext.Provider, { value, children: /* @__PURE__ */ jsx(
      Element,
      {
        ...rest,
        ref,
        "data-open": isOpen,
        className: styles,
        "aria-disabled": disabled,
        children
      }
    ) });
  }
);
AccordionItem.displayName = "MaterialTailwind.AccordionItem";
var AccordionTrigger = React.forwardRef(({ as, className, children, ...rest }, ref) => {
  const Element = as || "button";
  const contextTheme = useTheme();
  const theme = contextTheme.accordionTrigger || accordionTriggerTheme;
  const value = React.useContext(AccordionItemContext);
  const { type, activeItem, setActiveItem } = React.useContext(AccordionContext);
  const isMultiple = type === "multiple";
  const isOpen = isMultiple ? activeItem?.includes(value) : activeItem === value;
  const styles = twMerge(theme.baseStyle, className);
  return /* @__PURE__ */ jsx(
    Element,
    {
      ...rest,
      ref,
      "data-open": isOpen,
      className: styles,
      onClick: (event) => {
        if (isMultiple) {
          if (activeItem?.includes(value)) {
            setActiveItem?.(
              (prev) => prev.filter((item) => item !== value)
            );
          } else {
            setActiveItem?.((prev) => [...prev, value]);
          }
        } else {
          setActiveItem?.((prev) => prev === value ? "" : value);
        }
        rest.onClick?.(event);
      },
      children
    }
  );
});
AccordionTrigger.displayName = "MaterialTailwind.AccordionTrigger";
var AccordionContent = React.forwardRef(({ as, className, children, ...rest }, ref) => {
  const Element = as || "div";
  const contextTheme = useTheme();
  const theme = contextTheme.accordionContent || accordionContentTheme;
  const value = React.useContext(AccordionItemContext);
  const { type, activeItem } = React.useContext(AccordionContext);
  const isMultiple = type === "multiple";
  const isOpen = isMultiple ? activeItem?.includes(value) : activeItem === value;
  const styles = twMerge(theme.baseStyle, className);
  return isOpen ? /* @__PURE__ */ jsx(Element, { ...rest, ref, className: styles, "data-open": isOpen, children }) : null;
});
AccordionContent.displayName = "MaterialTailwind.AccordionContent";
var Accordion = Object.assign(AccordionRoot, {
  Item: AccordionItem,
  Trigger: AccordionTrigger,
  Content: AccordionContent
});
var accordion_default = Accordion;

export {
  AccordionContext,
  AccordionRoot,
  AccordionItemContext,
  AccordionItem,
  AccordionTrigger,
  AccordionContent,
  Accordion,
  accordion_default
};
