// @material-tailwind/react/chunk-BELULML5.js
import {
  useTheme
} from "./chunk-KYDE6FLK.js";
import {
  breadcrumbLinkTheme,
  breadcrumbSeparatorTheme,
  breadcrumbTheme
} from "./chunk-FOAJJTIZ.js";

// src/components/breadcrumb.tsx
import * as React from "react";
import { twMerge } from "tailwind-merge";
import { jsx } from "react/jsx-runtime";
var BreadcrumbRoot = React.forwardRef(
  ({ as, className, children, ...rest }, ref) => {
    const Element = as || "nav";
    const contextTheme = useTheme();
    const theme = contextTheme.breadcrumb || breadcrumbTheme;
    const styles = twMerge(theme.baseStyle, className);
    return /* @__PURE__ */ jsx(Element, { ...rest, ref, className: styles, children });
  }
);
BreadcrumbRoot.displayName = "MaterialTailwind.Breadcrumb";
var BreadcrumbLink = React.forwardRef(({ as, className, children, ...rest }, ref) => {
  const Element = as || "a";
  const contextTheme = useTheme();
  const theme = contextTheme.breadcrumbLink || breadcrumbLinkTheme;
  const styles = twMerge(theme.baseStyle, className);
  return /* @__PURE__ */ jsx(Element, { ...rest, ref, className: styles, children });
});
BreadcrumbLink.displayName = "MaterialTailwind.BreadcrumbLink";
var BreadcrumbSeparator = React.forwardRef(({ as, className, children, ...rest }, ref) => {
  const Element = as || "span";
  const contextTheme = useTheme();
  const theme = contextTheme.breadcrumbSeparator || breadcrumbSeparatorTheme;
  const styles = twMerge(theme.baseStyle, className);
  return /* @__PURE__ */ jsx(Element, { ...rest, ref, className: styles, children: children || "/" });
});
BreadcrumbSeparator.displayName = "MaterialTailwind.BreadcrumbSeparator";
var Breadcrumb = Object.assign(BreadcrumbRoot, {
  Link: BreadcrumbLink,
  Separator: BreadcrumbSeparator
});
var breadcrumb_default = Breadcrumb;

export {
  BreadcrumbRoot,
  BreadcrumbLink,
  BreadcrumbSeparator,
  Breadcrumb,
  breadcrumb_default
};
