// @material-tailwind/react/chunk-SZQANRDW.js
import {
  timelineBodyTheme,
  timelineHeaderTheme,
  timelineIconTheme,
  timelineItemTheme,
  timelineSeparatorTheme,
  timelineTheme
} from "./chunk-BAFDZUY3.js";
import {
  tooltipArrowTheme,
  tooltipContentTheme,
  tooltipTheme,
  tooltipTriggerTheme
} from "./chunk-CZOICNI5.js";
import {
  typographyTheme
} from "./chunk-36U2OMN6.js";
import {
  radioIndicatorTheme,
  radioItemTheme,
  radioTheme
} from "./chunk-R6IR6QGO.js";
import {
  ratingTheme
} from "./chunk-BGYX7VND.js";
import {
  selectListTheme,
  selectOptionTheme,
  selectTheme,
  selectTriggerTheme
} from "./chunk-TQ6RMPVT.js";
import {
  spinnerTheme
} from "./chunk-X5YG7IHD.js";
import {
  switchTheme
} from "./chunk-UCVBMWZ7.js";
import {
  tabsListTheme,
  tabsPanelTheme,
  tabsTheme,
  tabsTriggerIndicatorTheme,
  tabsTriggerTheme
} from "./chunk-N5SHDOJU.js";
import {
  textareaTheme
} from "./chunk-YPEJCH23.js";
import {
  drawerDismissTriggerTheme,
  drawerOverlayTheme,
  drawerPanelTheme,
  drawerTriggerTheme
} from "./chunk-KLVY2RS4.js";
import {
  iconButtonTheme
} from "./chunk-CO7SFVTO.js";
import {
  inputFieldTheme,
  inputIconTheme,
  inputTheme
} from "./chunk-FC6EJMVK.js";
import {
  listItemEndTheme,
  listItemStartTheme,
  listItemTheme,
  listTheme
} from "./chunk-PV5URN7B.js";
import {
  menuContentTheme,
  menuItemTheme,
  menuTheme,
  menuTriggerTheme
} from "./chunk-WZOFG7WK.js";
import {
  popoverArrowTheme,
  popoverContentTheme,
  popoverTheme,
  popoverTriggerTheme
} from "./chunk-CVARF25W.js";
import {
  progressBarTheme,
  progressTheme
} from "./chunk-7J5EVDFB.js";
import {
  breadcrumbLinkTheme,
  breadcrumbSeparatorTheme,
  breadcrumbTheme
} from "./chunk-FOAJJTIZ.js";
import {
  buttonGroupTheme
} from "./chunk-LN4CMBOV.js";
import {
  buttonTheme
} from "./chunk-LKROSG3R.js";
import {
  cardBodyTheme,
  cardFooterTheme,
  cardHeaderTheme,
  cardTheme
} from "./chunk-ELQB6BWL.js";
import {
  checkboxIndicatorTheme,
  checkboxTheme
} from "./chunk-QZKUZIYH.js";
import {
  chipDismissTriggerTheme,
  chipIconTheme,
  chipLabelTheme,
  chipTheme
} from "./chunk-GTV44CC2.js";
import {
  collapseTheme
} from "./chunk-4JMI2FIT.js";
import {
  dialogContentTheme,
  dialogDismissTriggerTheme,
  dialogOverlayTheme,
  dialogTheme,
  dialogTriggerTheme
} from "./chunk-DWFTZS5L.js";
import {
  accordionContentTheme,
  accordionItemTheme,
  accordionTheme,
  accordionTriggerTheme
} from "./chunk-T63Y5BBA.js";
import {
  alertContentTheme,
  alertDismissTriggerTheme,
  alertIconTheme,
  alertTheme
} from "./chunk-PMPU4UE3.js";
import {
  avatarTheme
} from "./chunk-PYSKRWL6.js";

// src/theme/theme.ts
var theme = {
  accordion: accordionTheme,
  accordionItem: accordionItemTheme,
  accordionTrigger: accordionTriggerTheme,
  accordionContent: accordionContentTheme,
  alert: alertTheme,
  alertContent: alertContentTheme,
  alertIcon: alertIconTheme,
  alertDismissTrigger: alertDismissTriggerTheme,
  avatar: avatarTheme,
  breadcrumb: breadcrumbTheme,
  breadcrumbLink: breadcrumbLinkTheme,
  breadcrumbSeparator: breadcrumbSeparatorTheme,
  button: buttonTheme,
  buttonGroup: buttonGroupTheme,
  checkbox: checkboxTheme,
  checkboxIndicator: checkboxIndicatorTheme,
  card: cardTheme,
  cardHeader: cardHeaderTheme,
  cardBody: cardBodyTheme,
  cardFooter: cardFooterTheme,
  chip: chipTheme,
  chipLabel: chipLabelTheme,
  chipIcon: chipIconTheme,
  chipDismissTrigger: chipDismissTriggerTheme,
  collapse: collapseTheme,
  dialog: dialogTheme,
  dialogTrigger: dialogTriggerTheme,
  dialogOverlay: dialogOverlayTheme,
  dialogContent: dialogContentTheme,
  dialogDismissTrigger: dialogDismissTriggerTheme,
  drawerTrigger: drawerTriggerTheme,
  drawerOverlay: drawerOverlayTheme,
  drawerPanel: drawerPanelTheme,
  drawerDismissTrigger: drawerDismissTriggerTheme,
  iconButton: iconButtonTheme,
  input: inputTheme,
  inputField: inputFieldTheme,
  inputIcon: inputIconTheme,
  list: listTheme,
  listItem: listItemTheme,
  listItemStart: listItemStartTheme,
  listItemEnd: listItemEndTheme,
  menu: menuTheme,
  menuContent: menuContentTheme,
  menuTrigger: menuTriggerTheme,
  menuItem: menuItemTheme,
  popover: popoverTheme,
  popoverTrigger: popoverTriggerTheme,
  popoverContent: popoverContentTheme,
  popoverArrow: popoverArrowTheme,
  progress: progressTheme,
  progressBar: progressBarTheme,
  radio: radioTheme,
  radioItem: radioItemTheme,
  radioIndicator: radioIndicatorTheme,
  rating: ratingTheme,
  select: selectTheme,
  selectTrigger: selectTriggerTheme,
  selectList: selectListTheme,
  selectOption: selectOptionTheme,
  spinner: spinnerTheme,
  switch: switchTheme,
  tabs: tabsTheme,
  tabsList: tabsListTheme,
  tabsTrigger: tabsTriggerTheme,
  tabsPanel: tabsPanelTheme,
  tabsTriggerIndicator: tabsTriggerIndicatorTheme,
  textarea: textareaTheme,
  tooltip: tooltipTheme,
  tooltipTrigger: tooltipTriggerTheme,
  tooltipContent: tooltipContentTheme,
  tooltipArrow: tooltipArrowTheme,
  typography: typographyTheme,
  timeline: timelineTheme,
  timelineItem: timelineItemTheme,
  timelineHeader: timelineHeaderTheme,
  timelineIcon: timelineIconTheme,
  timelineSeparator: timelineSeparatorTheme,
  timelineBody: timelineBodyTheme
};
var theme_default = theme;

export {
  theme,
  theme_default
};
