// @material-tailwind/react/theme/alert.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/theme/alert.ts
var alert_exports = {};
__export(alert_exports, {
  alertContentTheme: () => alertContentTheme,
  alertDismissTriggerTheme: () => alertDismissTriggerTheme,
  alertIconTheme: () => alertIconTheme,
  alertTheme: () => alertTheme
});
module.exports = __toCommonJS(alert_exports);
var alertTheme = {
  defaultProps: {
    size: "md",
    color: "primary",
    variant: "solid",
    rounded: false
  },
  baseStyle: "relative flex items-start w-full border rounded-md p-2 data-[pill=true]:rounded-full",
  variant: {
    ghost: {
      primary: "bg-primary/10 border-transparent text-primary",
      secondary: "bg-secondary/10 border-transparent text-secondary-foreground",
      info: "bg-info/10 border-transparent text-info",
      success: "bg-success/10 border-transparent text-success",
      warning: "bg-warning/10 border-transparent text-warning",
      error: "bg-error/10 border-transparent text-error"
    },
    solid: {
      primary: "bg-primary border-primary text-primary-foreground",
      secondary: "bg-secondary border-secondary text-secondary-foreground",
      info: "bg-info border-info text-info-foreground",
      success: "bg-success border-success text-success-foreground",
      warning: "bg-warning border-warning text-warning-foreground",
      error: "bg-error border-error text-error-foreground"
    },
    outline: {
      primary: "bg-transparent border-primary text-primary",
      secondary: "bg-transparent border-secondary text-secondary-foreground",
      info: "bg-transparent border-info text-info",
      success: "bg-transparent border-success text-success",
      warning: "bg-transparent border-warning text-warning",
      error: "bg-transparent border-error text-error"
    },
    gradient: {
      primary: "bg-gradient-to-tr from-primary-dark to-primary-light border-primary text-primary-foreground",
      secondary: "bg-gradient-to-tr from-secondary-dark to-secondary-light border-secondary text-secondary-foreground",
      info: "bg-gradient-to-tr from-info-dark to-info-light border-info text-info-foreground",
      success: "bg-gradient-to-tr from-success-dark to-success-light border-success text-success-foreground",
      warning: "bg-gradient-to-tr from-warning-dark to-warning-light border-warning text-warning-foreground",
      error: "bg-gradient-to-tr from-error-dark to-error-light border-error text-error-foreground"
    }
  }
};
var alertContentTheme = {
  baseStyle: "w-full text-base font-sans  leading-none m-1.5"
};
var alertIconTheme = {
  baseStyle: "grid place-items-center shrink-0 p-1"
};
var alertDismissTriggerTheme = {
  baseStyle: "outline-none"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  alertContentTheme,
  alertDismissTriggerTheme,
  alertIconTheme,
  alertTheme
});
