// @material-tailwind/react/theme/list.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/theme/list.ts
var list_exports = {};
__export(list_exports, {
  listItemEndTheme: () => listItemEndTheme,
  listItemStartTheme: () => listItemStartTheme,
  listItemTheme: () => listItemTheme,
  listTheme: () => listTheme
});
module.exports = __toCommonJS(list_exports);
var listTheme = {
  baseStyle: "flex flex-col gap-0.5 min-w-[240px]"
};
var listItemTheme = {
  defaultProps: {
    ripple: true
  },
  baseStyle: "flex items-center py-1.5 px-2.5 rounded-md align-middle select-none font-sans  transition-all duration-300 ease-in aria-disabled:opacity-50 aria-disabled:pointer-events-none bg-transparent text-foreground hover:text-black dark:hover:text-white hover:bg-surface focus:bg-surface focus:text-black dark:focus:text-white data-[selected=true]:bg-surface data-[selected=true]:text-black data-[selected=true]:text-white dark:bg-opacity-70"
};
var listItemStartTheme = {
  baseStyle: "grid place-items-center shrink-0 me-2.5"
};
var listItemEndTheme = {
  baseStyle: "grid place-items-center shrink-0 ps-2.5 ms-auto"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  listItemEndTheme,
  listItemStartTheme,
  listItemTheme,
  listTheme
});
