// @material-tailwind/react/theme/menu.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/theme/menu.ts
var menu_exports = {};
__export(menu_exports, {
  menuContentTheme: () => menuContentTheme,
  menuItemTheme: () => menuItemTheme,
  menuTheme: () => menuTheme,
  menuTriggerTheme: () => menuTriggerTheme
});
module.exports = __toCommonJS(menu_exports);
var menuTheme = {
  defaultProps: {
    placement: "bottom",
    offset: 5
  }
};
var menuTriggerTheme = {
  baseStyle: "outline-none group"
};
var menuContentTheme = {
  defaultProps: {
    disabled: false,
    initialFocus: 0,
    returnFocus: true,
    guards: true,
    modal: false,
    visuallyHiddenDismiss: true,
    closeOnFocusOut: true,
    order: ["content"]
  },
  baseStyle: "min-w-[160px] rounded-lg space-y-0.5 border border-surface bg-background p-1 text-sm text-foreground shadow-xl shadow-black/[0.025] outline-none"
};
var menuItemTheme = {
  defaultProps: {
    ripple: true,
    closeOnClick: true
  },
  baseStyle: "w-full text-start flex items-center py-1.5 px-2.5 rounded-[5px] align-middle select-none outline-none font-sans  transition-all duration-300 ease-in aria-disabled:opacity-50 aria-disabled:pointer-events-none bg-transparent text-foreground hover:text-black dark:hover:text-white hover:bg-surface focus:bg-surface focus:text-black dark:focus:text-white data-[selected=true]:bg-surface data-[selected=true]:text-black dark:data-[selected=true]:text-white dark:bg-opacity-70"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  menuContentTheme,
  menuItemTheme,
  menuTheme,
  menuTriggerTheme
});
