// @material-tailwind/react/theme/switch.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/theme/switch.ts
var switch_exports = {};
__export(switch_exports, {
  switchTheme: () => switchTheme
});
module.exports = __toCommonJS(switch_exports);
var switchTheme = {
  defaultProps: {
    color: "primary"
  },
  baseStyle: "appearance-none relative inline-block rounded-full w-10 h-6 before:inline-block before:absolute before:top-0 before:left-0 before:w-full before:h-full before:rounded-full before:bg-surface before:transition-colors before:duration-200 before:ease-in after:absolute after:top-2/4 after:left-0.5 after:-translate-y-2/4 after:w-5 after:h-5 after:bg-background after:rounded-full checked:after:translate-x-[calc(100%-4px)] after:transition-transform after:duration-200 after:ease-in disabled:opacity-50 disabled:cursor-not-allowed dark:after:bg-white",
  color: {
    primary: "checked:before:bg-primary",
    secondary: "checked:before:bg-secondary",
    info: "checked:before:bg-info",
    success: "checked:before:bg-success",
    warning: "checked:before:bg-warning",
    error: "checked:before:bg-error"
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  switchTheme
});
