// @material-tailwind/react/theme/tabs.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/theme/tabs.ts
var tabs_exports = {};
__export(tabs_exports, {
  tabsListTheme: () => tabsListTheme,
  tabsPanelTheme: () => tabsPanelTheme,
  tabsTheme: () => tabsTheme,
  tabsTriggerIndicatorTheme: () => tabsTriggerIndicatorTheme,
  tabsTriggerTheme: () => tabsTriggerTheme
});
module.exports = __toCommonJS(tabs_exports);
var tabsTheme = {
  defaultProps: {
    orientation: "horizontal"
  },
  baseStyle: "flex data-[orientation=horizontal]:flex-col data-[orientation=vertical]:flex-row gap-2"
};
var tabsListTheme = {
  baseStyle: "flex shrink-0 w-max data-[orientation=horizontal]:flex-row data-[orientation=vertical]:flex-col rounded-md p-1 bg-surface-light dark:bg-surface"
};
var tabsTriggerTheme = {
  baseStyle: "inline-flex relative z-[2] py-1.5 px-4 items-center justify-center align-middle text-black dark:text-white select-none font-sans font-medium text-center aria-disabled:opacity-50 aria-disabled:pointer-events-none"
};
var tabsPanelTheme = {
  baseStyle: "p-1 w-full block"
};
var tabsTriggerIndicatorTheme = {
  baseStyle: "bg-background rounded shadow-sm shadow-black/10 transition-all duration-300 ease-in"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  tabsListTheme,
  tabsPanelTheme,
  tabsTheme,
  tabsTriggerIndicatorTheme,
  tabsTriggerTheme
});
