// @material-tailwind/react/theme/tooltip.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/theme/tooltip.ts
var tooltip_exports = {};
__export(tooltip_exports, {
  tooltipArrowTheme: () => tooltipArrowTheme,
  tooltipContentTheme: () => tooltipContentTheme,
  tooltipTheme: () => tooltipTheme,
  tooltipTriggerTheme: () => tooltipTriggerTheme
});
module.exports = __toCommonJS(tooltip_exports);
var tooltipTheme = {
  defaultProps: {
    placement: "top",
    offset: 8,
    interactive: false
  }
};
var tooltipTriggerTheme = {
  baseStyle: "outline-none group"
};
var tooltipContentTheme = {
  baseStyle: "rounded-md bg-black dark:bg-white border-black dark:border-white px-1.5 pt-1 pb-[3px] text-xs text-white dark:text-black shadow-lg shadow-black/5 outline-none"
};
var tooltipArrowTheme = {
  baseStyle: "h-2 w-2 rounded-bl-[3px] border border-[inherit] bg-inherit [clip-path:polygon(0_0,_100%_100%,_0_100%)] data-[placement^=bottom]:rotate-[135deg] data-[placement^=left]:rotate-[225deg] data-[placement^=right]:rotate-45 data-[placement^=top]:-rotate-45"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  tooltipArrowTheme,
  tooltipContentTheme,
  tooltipTheme,
  tooltipTriggerTheme
});
